primo/opt_model/result_parser.py

```diff
@@ -0,0 +1,800 @@
+                # these are hardcoded...they need to be unchangeable
+                max_value = 5
+                min_value = 0
+            elif metric.name == "num_unique_owners":
+                metric.data_col_name = metric.name
+                max_value = self.campaign.get_max_value_across_all_projects(metric.name)
+                min_value = 1
+            elif metric.name == "num_wells":
+                metric.data_col_name = metric.name
+                max_value = self.campaign.get_max_value_across_all_projects(metric.name)
+                min_value = 1
+            else:
+                max_value = self.campaign.get_max_value_across_all_wells(
+                    metric.data_col_name
+                )
+                min_value = self.campaign.get_min_value_across_all_wells(
+                    metric.data_col_name
+                )
+            assert getattr(project, metric.score_attribute, None) is None
+            assert hasattr(project, metric.name)
+
+            # Check if division by a zero is likely
+            if np.isclose(max_value, min_value, rtol=0.001):
+                # All cells in this column have equal value.
+                # To avoid division by zero, set min_value = 0
+                min_value = 0
+
+            if np.isclose(max_value, 0, rtol=0.001):
+                # All values in this column are likely zeros.
+                # To avoid division by zero, set max_value = 1
+                max_value = 1.0
+
+            if metric.has_inverse_priority:
+                setattr(
+                    project,
+                    metric.score_attribute,
+                    (
+                        (max_value - getattr(project, metric.name))
+                        / (max_value - min_value)
+                    )
+                    * metric.effective_weight,
+                )
+
+            else:
+                setattr(
+                    project,
+                    metric.score_attribute,
+                    (
+                        (getattr(project, metric.name) - min_value)
+                        / (max_value - min_value)
+                    )
+                    * metric.effective_weight,
+                )
+
+    def compute_overall_efficiency_scores_project(self, project):
+        """
+        Computes the overall efficiency score for a project
+
+        Parameters
+        ----------
+        project : OptimalProject
+            project in an OptimalCampaign
+        """
+        names_attributes = [i for i in dir(project) if "eff_score" in i]
+        assert len(names_attributes) == len(
+            [
+                metric.name
+                for metric in self.efficiency_weights
+                if metric.weight != 0 and not hasattr(metric, "submetric")
+            ]
+        )
+        for attr in names_attributes:
+            project.update_efficiency_score(getattr(project, attr))
+
+    def compute_overall_efficiency_scores_campaign(self):

```

Comment:
It might be helpful to log more details here about which efficiency score is being updated

Reply:
I do use logging to tell the user which efficiency score is being computed on (what is currently) line 666. The part of the code with your comment computes the overall efficiency score of each project. I added a logging message here to say "Computing overall efficiency score". Does that address your comment satisfactorily?